python/cudf/cudf/utils/dtypes.py

```diff
@@ -671,6 +672,43 @@ def _maybe_convert_to_default_type(dtype):
     return dtype
 
 
+def _dtype_can_hold_range(rng: range, dtype: np.dtype) -> bool:
+    if not len(rng):
+        return True
+    return np.can_cast(rng[0], dtype) and np.can_cast(rng[-1], dtype)
+
+
+def _numeric_can_hold_element(dtype: np.dtype, element):
```

Comment:
Should this be named `_dtype_can_hold_element()` instead, for parity with `_dtype_can_hold_range()`?

The `numeric` seems unnecessary unless we _never_ want to support non-numeric types here.

Reply:
Yup, renamed it 👍